subsys/bluetooth/host/scan.c

```diff
@@ -1187,6 +1187,11 @@ void bt_hci_le_adv_report(struct net_buf *buf)
 
 		evt = net_buf_pull_mem(buf, sizeof(*evt));
 
+		if (buf->len <= evt->length) {
```

Comment:
It's maybe a bit too subtle that you use `<=` instead of `<` to cover for the one octet RSSI value that's required at the end. `if (buf->len < evt->length + sizeof(adv_info.rssi))` would be clearer.